### PR TITLE
feat: Add session recipe

### PR DESCRIPTION
**Pull request by nkshah2**
## Summary
- Adds Session recipe
- Adds helper functions for Session recipe to authRecipe wrapper
- Exports Session helper functions from emailpassword, thirdparty and thirdpartyemailpassword
- Updates Size limits (Note: the size has gone up significantly because supertokens-website is now added as a dependency and all recipes except emailverification use the Session recipe)

## Remaining TODOs
- [x] Update size limit
- [x] Add default export for Session recipe

### CODE REVIEW

File: lib/ts/recipe/authRecipe/index.ts [hunk 1]
```diff
@@ -24,4 +25,16 @@ export default abstract class AuthRecipe<
...
+        return await Session.getInstanceOrThrow().signOut({
+            userContext: input.userContext,
+        });
+    };
+
+    doesSessionExist = async (input: { userContext: any }): Promise<boolean> => {
+        return await Session.getInstanceOrThrow().doesSessionExist({
+            userContext: input.userContext,
+        });
+    };
```
Comment (rishabhpoddar) [inline lib/ts/recipe/authRecipe/index.ts:39]: Remove this.

File: lib/ts/recipe/emailpassword/index.ts [hunk 2]
```diff
@@ -23,6 +23,18 @@ export default class RecipeWrapper {
...
+        return Recipe.getInstanceOrThrow().signOut({
+            userContext: getNormalisedUserContext(input?.userContext),
+        });
+    }
+
+    static doesSessionExist(input: { userContext: any }): Promise<boolean> {
+        return Recipe.getInstanceOrThrow().doesSessionExist({
+            userContext: input.userContext,
+        });
+    }
```
Comment (rishabhpoddar) [inline lib/ts/recipe/emailpassword/index.ts:36]: Don't need this from any of the auth recieps.

File: lib/ts/recipe/emailverification/index.ts [hunk 3]
```diff
@@ -18,7 +18,7 @@ import { RecipeFunctionOptions } from "../recipeModule/types";
 import { getNormalisedUserContext } from "../../utils";
 
 export default class RecipeWrapper {
-    static init(config?: InputType) {
+    static init(config: InputType) {
```
Comment (rishabhpoddar) [inline lib/ts/recipe/emailverification/index.ts:21]:
```suggestion
    static init(config?: InputType) {
```

File: lib/ts/recipe/emailverification/recipe.ts [hunk 4]
```diff
@@ -41,7 +41,7 @@ export default class Recipe implements RecipeModule<PreAndPostAPIHookAction, Nor
         this.recipeImplementation = builder.override(this.config.override.functions).build();
     }
 
-    static init(config?: InputType): CreateRecipeFunction<PreAndPostAPIHookAction> {
+    static init(config: InputType): CreateRecipeFunction<PreAndPostAPIHookAction> {
```
Comment (rishabhpoddar) [inline lib/ts/recipe/emailverification/recipe.ts:44]:
```suggestion
    static init(config?: InputType): CreateRecipeFunction<PreAndPostAPIHookAction> {
```

File: lib/ts/recipe/session/types.ts [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+    sessionExpiredStatusCode?: number;
+    autoAddCredentials?: boolean;
+    isInIframe?: boolean;
+    cookieDomain?: string;
+    preAPIHook?: (context: {
+        action: "SIGN_OUT" | "REFRESH_SESSION";
+        requestInit: RequestInit;
+        url: string;
+    }) => Promise<{ url: string; requestInit: RequestInit }>;
+    onHandleEvent?: (event: RecipeEvent) => void;
```
Comment (rishabhpoddar) [inline lib/ts/recipe/session/types.ts:45]: Also add a postAPIHook (which will also change supertokens-website)

File: lib/ts/recipe/thirdpartyemailpassword/index.ts [hunk 6]
```diff
@@ -23,6 +23,18 @@ export default class RecipeWrapper {
         return Recipe.init(config);
     }
 
+    static signOut(input?: { userContext?: any }) {
+        return Recipe.getInstanceOrThrow().signOut({
+            userContext: getNormalisedUserContext(input?.userContext),
```
Comment (rishabhpoddar) [inline lib/ts/recipe/thirdpartyemailpassword/index.ts:28]: Please make sure the tsconfig is the exact same as that of supertokens-website. So things like using "?" won't work

**Review by rishabhpoddar — CHANGES_REQUESTED** [review]

File: lib/ts/recipe/thirdpartyemailpassword/index.ts (hunk 6 above)
Comment (nkshah2) [reply lib/ts/recipe/thirdpartyemailpassword/index.ts:28]: Moved as a point in the checklist here: https://github.com/supertokens/for-zenhub/issues/28

**Review by rishabhpoddar — APPROVED** [review]

**Review by rishabhpoddar — APPROVED** [review]

**Pull request merged**